feat: cloudant da

rajatagarwal-ibm opened the pull request:
### Description

Creating DA for cloudant deployed FSCloud compliant 

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

rajatagarwal-ibm commented:
/run pipeline

ocofaigh commented on tests/pr_test.go:
```diff
@@ -61,21 +63,29 @@ func TestRunUpgradeExample(t *testing.T) {
+		"access_tags":      permanentResources["accessTags"],
+		// crn of the dedicated host
+		"environment_crn":   permanentResources["dedicatedHostCrn"],
+		"resource_group_id": permanentResources["resourceGroupTestPermanentId"],
```
We don't want to use this resource group (this is for permanent resources only).. We should use the `resourceGroup` const we have defined in pr_test.go

ocofaigh commented on tests/pr_test.go:
```diff
@@ -61,21 +63,29 @@ func TestRunUpgradeExample(t *testing.T) {
+	// Generate a 6 char long random string for the instance_name
+	randomBytes := make([]byte, 6)
+	_, err := rand.Read(randomBytes)
+	randomInstanceName := "fscloud" + base64.URLEncoding.EncodeToString(randomBytes)[:6]
```
Do we need this? Can we not pass `Prefix: "fscloud",` into the setup options, and then reference it like options.Prefix ? I think it should then return a randomly generated prefix value?

ocofaigh commented on solutions/fscloud/version.tf:
```diff
@@ -0,0 +1,14 @@
+      source = "IBM-Cloud/ibm"
+      # Use "greater than or equal to" range in modules
+      version = ">= 1.56.1, < 2.0.0"
+    }
```
A DA should always lock into an exact version of any required providers. 

Why do you have the `tflint-ignore` ? Does the linter actually fail if you remove it??

ocofaigh commented on solutions/fscloud/README.md:
```diff
@@ -0,0 +1,58 @@
+# Profile for IBM Cloud Framework for Financial Services
```
This isn't a profile or submodule. Its a DA (Deployable Architecture), so we need to come up with a better title. This will show in the catalog. Will need @SirSpidey input here

ocofaigh replied:
Same goes for everything we have in the doc here.

ocofaigh commented on solutions/fscloud/README.md:
```diff
@@ -0,0 +1,58 @@
+# Profile for IBM Cloud Framework for Financial Services
+
+This code is a version of the [parent root module](../../) that includes a default configuration that complies with the relevant controls from the [IBM Cloud Framework for Financial Services](https://cloud.ibm.com/docs/framework-financial-services?topic=framework-financial-services-about). See the [Example for IBM Cloud Framework for Financial Services](/examples/fscloud/) for logic that uses thi ...
```
Remove this for sure

ocofaigh commented on solutions/fscloud/README.md:
```diff
@@ -0,0 +1,58 @@
+
+This code is a version of the [parent root module](../../) that includes a default configuration that complies with the relevant controls from the [IBM Cloud Framework for Financial Services](https://cloud.ibm.com/docs/framework-financial-services?topic=framework-financial-services-about). See the [Example for IBM Cloud Framework for Financial Services](/examples/fscloud/) for logic that uses thi ...
+
+:exclamation: **Exception:** The Cloudant DB service is not yet Financial services validated. Therefore, the infrastructure that is deployed by this module is also not validated with the Framework for Financial Services. For more information, see the list of [Financial Services Validated services]( https://cloud.ibm.com/docs/framework-financial-services?topic=framework-financial-services-vpc-arch ...
```
I dont think we will call the DA fscloud, so remove this

ocofaigh commented on solutions/fscloud/README.md:
```diff
@@ -0,0 +1,58 @@
+
+:exclamation: **Exception:** The Cloudant DB service is not yet Financial services validated. Therefore, the infrastructure that is deployed by this module is also not validated with the Framework for Financial Services. For more information, see the list of [Financial Services Validated services]( https://cloud.ibm.com/docs/framework-financial-services?topic=framework-financial-services-vpc-arch ...
+
+The default values in this profile were scanned by [IBM Code Risk Analyzer (CRA)](https://cloud.ibm.com/docs/code-risk-analyzer-cli-plugin?topic=code-risk-analyzer-cli-plugin-cra-cli-plugin#terraform-command) for compliance with the IBM Cloud Framework for Financial Services profile that is specified by the IBM Security and Compliance Center. The scan passed for all applicable goals.
```
Remove this

ocofaigh commented on solutions/fscloud/README.md:
```diff
@@ -0,0 +1,58 @@
+The default values in this profile were scanned by [IBM Code Risk Analyzer (CRA)](https://cloud.ibm.com/docs/code-risk-analyzer-cli-plugin?topic=code-risk-analyzer-cli-plugin-cra-cli-plugin#terraform-command) for compliance with the IBM Cloud Framework for Financial Services profile that is specified by the IBM Security and Compliance Center. The scan passed for all applicable goals.
+
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+### Requirements
```
Need to decide if we want this to show in catalog or not

ocofaigh reviewed (changes requested):
I left some comments. I'm reluctant to call any of this fscloud since cloudant is not fscloud validated. We need to decide what the offering and flavour name will be in catalog. Do we call the offering "Cloudant DA" and call this flavour of it `secure` ?

@SirSpidey will need your help on this.

ocofaigh reviewed (commented):
I don’t think the DA code should be in a named directory named `fscloud` - I suggest we name the directroy the same name as the DA flavour (which we will decide on a call later today)

ocofaigh commented on solutions/fscloud/README.md:
```diff
@@ -0,0 +1,55 @@
+# Cloudant Deployable Architecture (DA)
+
```
We need to add some description here as to what the DA provisions, and supports

ocofaigh commented on solutions/fscloud/main.tf:
```diff
@@ -0,0 +1,22 @@
+module "resource_group" {
+  count                        = (var.resource_group_id == null) ? 1 : 0
```
You don't need a count here - this module will always run. It will create a new RG is `existing_resource_group` is `false` or it will return the RG ID if `existing_resource_group` is `true`

ocofaigh commented on solutions/fscloud/main.tf:
```diff
@@ -0,0 +1,22 @@
+
+module "cloudant" {
+  source              = "../../modules/fscloud"
+  resource_group_id   = var.resource_group_id == null ? module.resource_group[0].resource_group_id : var.resource_group_id
```
Remove the count ad per https://github.com/terraform-ibm-modules/terraform-ibm-cloudant/pull/32/files#r1392398763 and always grab this value from the module

rajatagarwal-ibm replied:
If I have to remove `count` then the `resource_group_id` variable will be removed too. That means the User has only 2 options - either to pass the existing resource group name or not pass anything (in that case, it will create a new resource group). There will not be any provision to pass "resource group ID" anymore. If we are okay with this design I will implement the changes.

ocofaigh replied:
yep - thats what we want

ocofaigh commented on solutions/fscloud/variables.tf:
```diff
@@ -0,0 +1,74 @@
+  default     = "us-south"
+}
+
+variable "resource_group_id" {
```
This is not needed, we will use `resource_group_name` instead

ocofaigh commented on solutions/fscloud/variables.tf:
```diff
@@ -0,0 +1,74 @@
+
+variable "tags" {
+  type        = list(string)
+  description = "List of tags to be associated to cloudant instance"
```
List of resource tags

ocofaigh commented on solutions/fscloud/variables.tf:
```diff
@@ -0,0 +1,74 @@
+variable "include_data_events" {
+  type        = bool
+  description = "Include data event types in events sent to IBM Cloud Activity Tracker. If set to false, only management events will be sent to Activity Tracker."
+  default     = false
```
Do you know why we default to false here? Wondering if the better practise is to default to `true`. What do the docs recommend? What will Dreadnought be doing?

ocofaigh commented on tests/pr_test.go:
```diff
@@ -61,21 +61,27 @@ func TestRunUpgradeExample(t *testing.T) {
+		"environment_crn":         permanentResources["dedicatedHostCrn"],
+		"existing_resource_group": true,
+		"resource_group_name":     options.ResourceGroup,
+		"instance_name":           options.Prefix,
```
We probably want to pass a value for `database_config` here in order to test database creation

ocofaigh reviewed (changes requested):
As discussed, lets call the folder `dedicated` instead of `fscloud`

rajatagarwal-ibm commented:
/run pipeline

ocofaigh commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,60 @@
+# Deployable Architecture (DA) for the dedicated IBM Cloudant instance
+
+Deployable Architecture for the dedicated IBM Cloudant instance which deploys the following resources:
```
which supports provisioning the following resources:

ocofaigh commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,60 @@
+Deployable Architecture for the dedicated IBM Cloudant instance which deploys the following resources:
+
+- A resource group, if one is not passed in.
+- A dedicated cloudant instance
```
An IBM Cloudant Dedicated instance in IBM Cloud Dedicated (seems strange, but thats what Cloudant [docs](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) have). @SirSpidey can advise

ocofaigh commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,60 @@
+
+- A resource group, if one is not passed in.
+- A dedicated cloudant instance
+- A basic database in the Cloudant instance
```
By default it doesn't create any database. It supports optionally creating one, so suggest saying:

`- A Cloudant database`

ocofaigh commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,60 @@
+- A resource group, if one is not passed in.
+- A dedicated cloudant instance
+- A basic database in the Cloudant instance
+
```
I think we need to add a Before you begin section:

```
## Before you begin

* You need an [IBM Cloudant Dedicated](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) Hardware plan instance in order to provision the Cloudant instance in
```

ocofaigh commented on solutions/dedicated/version.tf:
```diff
@@ -0,0 +1,10 @@
+  required_providers {
+    ibm = {
+      source = "IBM-Cloud/ibm"
+      # Use "greater than or equal to" range in modules
```
remove this code comment - this is not a module, its a DA, so we lock into exact version like you have

ocofaigh reviewed (changes requested):
See comments. Can you also update the renovate.json with this:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["github>terraform-ibm-modules/common-dev-assets:commonRenovateConfig"],
  "packageRules": [
    {
      "description": "Use fix as Semantic Commit prefix for all dependency updates in these folders to generate release",
      "matchPaths": ["solutions/dedicated"],
      "semanticCommitType": "fix"
    },
    {
      "description": "Allow the locked in provider version to be updated to the latest for DAs",
      "enabled": true,
      "matchPaths": ["solutions/dedicated"],
      "matchManagers": ["terraform"],
      "matchDepTypes": ["required_provider"],
      "rangeStrategy": "bump",
      "semanticCommitType": "fix",
      "group": true,
      "groupName": "required_provider",
      "commitMessageExtra": "to latest for the DA solution"
    }
  ]
}
```

SirSpidey reviewed (commented):
I don't know enough about Cloudant and Cloud Dedicated to suggest the correct naming. We might need to talk to someone in the Cloudant  team to get the proper naming. 

But I don't think we want to label this a DA, just as we don't in the landing zone repo.

The docs that this will need to be onboarded as a DA include: 
- a reference architecture: https://test.cloud.ibm.com/docs-internal/writing?topic=writing-reference-architectures. Includes a .md file and an SVG in a `/reference-architectures` directory
- an overview in the landing zone deployment guide: https://test.cloud.ibm.com/docs/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-overview
-

rajatagarwal-ibm commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,60 @@
+# Deployable Architecture (DA) for the dedicated IBM Cloudant instance
```
As per the docs:

> IBM Cloud Dedicated customers can also choose to use IBM Cloudant in IBM Cloud Public by using the syndicated catalog, including the Lite and Standard plan instances. IBM Cloud Dedicated users aren't eligible to buy the Dedicated Hardware plan from the IBM Cloud Public catalog. IBM Cloud Dedicated users must use IBM Cloudant Dedicated on IBM Cloud Dedicated if dedicated hardware is required.

I think the exact title should be `IBM Cloudant Dedicated on IBM Cloud Dedicated` as it is deployed on the dedicated hardware. I have changed the title to that. Thoughts? 

@ocofaigh @SirSpidey

SirSpidey replied:
It isn't a pretty title...
If we're not expecting another module of IBM Cloudant Dedicated on `something else`, I think `IBM Cloudant Dedicated` is good enough as the title. We can get into the ICD on ICD mesh of words in the description.

rajatagarwal-ibm commented:
> I don't know enough about Cloudant and Cloud Dedicated to suggest the correct naming. We might need to talk to someone in the Cloudant team to get the proper naming.
> 
> But I don't think we want to label this a DA, just as we don't in the landing zone repo.
> 
> The docs that this will need to be onboarded as a DA include:
> 
> * a reference architecture: https://test.cloud.ibm.com/docs-internal/writing?topic=writing-reference-architectures. Includes a .md file and an SVG in a `/reference-architectures` directory
> * an overview in the landing zone deployment guide: https://test.cloud.ibm.com/docs/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-overview

Got it, so in that case, are we keeping it like a landing zone and not calling this a DA? Which means not adding any reference architecture or deployment guide ie keeping is as is for now?

SirSpidey commented:
> Got it, so in that case, are we keeping it like a landing zone and not calling this a DA? This means not adding any reference architecture or deployment guide, ie keeping it as is for now.

Still need the ref arch and new content for the existing deployment guide for IBM Cloud. Just don't need to call it a DA here in the repo.

rajatagarwal-ibm commented:
/run pipeline

rajatagarwal-ibm commented:
> > Got it, so in that case, are we keeping it like a landing zone and not calling this a DA? This means not adding any reference architecture or deployment guide, ie keeping it as is for now.
> 
> Still need the ref arch and new content for the existing deployment guide for IBM Cloud. Just don't need to call it a DA here in the repo.

Removed "Deployable Architecture" and replaced it with Solutions in the first line of readme

rajatagarwal-ibm commented:
/run pipeline

SirSpidey commented:
Putting the pieces back together. Here's where I think we are after our naming discussion: 
- Title in root readme file here: `Cloudant module`
- Title of DA: `Cloudant`
- Flavor: `Dedicated`
    - Title in readme file in `solutions/dedicated`: IBM Cloudant Dedicated *on IBM Cloud Dedicated* (still not sure we need the `on ICD`)
    - Content in readme file in `solutions/dedicated`: Verify that we want the inputs and other information from the TERRAFORM DOCS HOOK -- it will display in the DA in IBM Cloud.

SirSpidey commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,64 @@
+# IBM Cloudant Dedicated on IBM Cloud Dedicated
+
+Solutions for an IBM Cloudant dedicated instance which supports provisioning the following resources:
```
```suggestion
This architecture creates an IBM Cloudant Dedicated instance in IBM Cloud Dedicated and provisions the following resources:
```

SirSpidey commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,64 @@
+Solutions for an IBM Cloudant dedicated instance which supports provisioning the following resources:
+
+- A resource group, if one is not passed in.
+- An IBM Cloudant Dedicated instance in IBM Cloud Dedicated
```
```suggestion
- An IBM Cloudant Dedicated instance on IBM Cloud Dedicated
```

SirSpidey commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,64 @@
+
+- A resource group, if one is not passed in.
+- An IBM Cloudant Dedicated instance in IBM Cloud Dedicated
+- A Cloudant database
```
```suggestion
- A IBM Cloudant database
```

SirSpidey commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,64 @@
+
+## Before you begin
+
+* You need an [IBM Cloudant Dedicated](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) Hardware plan instance in order to provision the Cloudant instance in.
```
```suggestion
* You need an IBM Cloudant [Dedicated Hardware plan instance](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) for the IBM Cloudant instance.
```

Using a Dedicated Hardware plan instance

SirSpidey commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,64 @@
+
+* You need an [IBM Cloudant Dedicated](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) Hardware plan instance in order to provision the Cloudant instance in.
+
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
This infor will appear with the DA in the cloud catalog, so we want to make sure that it makes sense in that context.

rajatagarwal-ibm replied:
Removing the pre-commit-terraform docs hook, we don't want to display this information (Same as in SLZ module).

SirSpidey reviewed (changes requested):
I think I finally understand what I'm reviewing. 😿

rajatagarwal-ibm commented:
/run pipeline

SirSpidey commented on solutions/dedicated/variables.tf:
```diff
@@ -0,0 +1,67 @@
+
+variable "include_data_events" {
+  type        = bool
+  description = "Include data event types in events sent to IBM Cloud Activity Tracker. If set to false, only management events will be sent to Activity Tracker."
+  default     = false
+}
+
+variable "capacity" {
+  type        = number
+  description = "Number of blocks of throughput units. See https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-public#provisioned-throughput-capacity."
+  default     = 1
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "List of access tags to be associated with the Cloudant instance"
+  default     = []
+}
+
+variable "tags" {
+  type        = list(string)
+  description = "List of resource tags to be associated to cloudant instance"
+  default     = []
+}
+
+variable "environment_crn" {
+  description = "CRN of the IBM Cloudant Dedicated Hardware plan instance to provision a cloudant instance"
+  type        = string
+}
+
+variable "database_config" {
+  type = list(object({
+    db          = string
+    partitioned = optional(bool)
+    shards      = optional(number)
+  }))
+
+  description = "(Optional, List) The databases with their corresponding partitioning and shards to be created in the cloudant instance"
+  default     = []
+}
```
```suggestion
variable "ibmcloud_api_key" {
  description = "The IBM Cloud API key to deploy IAM-enabled resources."
  type        = string
  sensitive   = true
}

variable "region" {
  type        = string
  description = "The IBM Cloud region where the instance of IBM Cloudant is provisioned."
  default     = "us-south"
}

variable "existing_resource_group" {
  type        = bool
  description = "Whether to use an existing resource group. If false, specify the name for the new resource group in the `resource_group_name` input."
  default     = false
}

variable "resource_group_name" {
  type        = string
  description = "Add a new resource group with this name if the `existing_resource_group` input is false. To use an existing resource group, set `existing_resource_group` to true instead."
}

variable "instance_name" {
  description = "The name of the IBM Cloudant instance."
  type        = string
}

variable "include_data_events" {
  type        = bool
  description = "Whether to include data event types in events that are sent to IBM Cloud Activity Tracker. If false, only management events are sent."
  default     = false
}

variable "capacity" {
  type        = number
  description = "The number of blocks of throughput units. See [Provisioned throughput capacity](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-public#provisioned-throughput-capacity)."
  default     = 1
}

variable "access_tags" {
  type        = list(string)
  description = "The list of access tags to apply to the IBM Cloudant instance."
  default     = []
}

variable "tags" {
  type        = list(string)
  description = "The list of resource tags to apply to The IBM Cloudant instance."
  default     = []
}

variable "environment_crn" {
  description = "The CRN of the instance of the IBM Cloudant Dedicated Hardware plan to provision an IBM Cloudant instance"
  type        = string
}

variable "database_config" {
  type = list(object({
    db          = string
    partitioned = optional(bool)
    shards      = optional(number)
  }))

  description = "The databases to create in the IBM Cloudant instance and whether they are partitioned (or the number of shards if non-partitioned)."
  default     = []
}
```

SirSpidey reviewed (changes requested):
Reviewed inputs and outputs for consistency and word usage because these strings appear in the Catalog UI. Check the description `database_config` for accuracy -- I assume the database is either non-partitioned and has shards or partitioned (and listed as true).

As far as I know, markdown links will render in the UI, so changed the one URL.

rajatagarwal-ibm commented:
/run pipeline

rajatagarwal-ibm commented:
/run pipeline

SirSpidey reviewed (approved):
thanks for checking variables across the repo

rajatagarwal-ibm commented:
/run pipeline

ocofaigh commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,11 @@
+# IBM Cloudant Dedicated on IBM Cloud Dedicated
+
+This architecture creates an IBM Cloudant Dedicated instance in IBM Cloud Dedicated and provisions the following resources:
```
I think this needs to be re-worded:
`provisions the following resources` -> `supports provisioning the following resources` otherwise it seems as if the DA will create a database by default, which is not the case (its an opt in feature)

ocofaigh commented on solutions/dedicated/main.tf:
```diff
@@ -0,0 +1,19 @@
+module "resource_group" {
+  source                       = "terraform-ibm-modules/resource-group/ibm"
+  version                      = "1.1.0"
```
1.1.4 is the latest

ocofaigh commented on solutions/dedicated/variables.tf:
```diff
@@ -0,0 +1,67 @@
+
+variable "resource_group_name" {
+  type        = string
+  description = "Add a new resource group with this name if the `existing_resource_group` input is false. To use an existing resource group, set `existing_resource_group` to true instead."
```
This description is inaccurate. This value is required regardless of whether `existing_resource_group` os true or false. Suggest to change to:
"The name of a new or an existing resource group in which to provision the Cloudant instance in."

ocofaigh commented on solutions/dedicated/variables.tf:
```diff
@@ -0,0 +1,67 @@
+
+variable "existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group. If false, specify the name for the new resource group in the `resource_group_name` input."
```
Remove this line: "If false, specify the name for the new resource group in the `resource_group_name` input."

`resource_group_name` is required regardless of whether this variable is true or false

ocofaigh commented on tests/pr_test.go:
```diff
@@ -61,21 +61,34 @@ func TestRunUpgradeExample(t *testing.T) {
 }
 
-func TestRunFSCloudExample(t *testing.T) {
+func TestRunDedicatedSolutions(t *testing.T) {
```
TestRunDedicatedSolutions -> TestRunDedicatedSolution

ocofaigh reviewed (changes requested):
See final few comments. We also need a diagram too, but that can come in a separate PR if you wish

ocofaigh commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,13 @@
+# IBM Cloudant Dedicated on IBM Cloud Dedicated
+
+This architecture creates an IBM Cloudant Dedicated instance in IBM Cloud Dedicated and supports provisioning the following resources:
```
I think this reads better: `in an IBM Cloud Dedicated environment`

rajatagarwal-ibm replied:
Can we leave it as is because not everything is in the dedicated environment - like resource group

ocofaigh replied:
I was suggesting: 

This architecture creates an IBM Cloudant Dedicated instance in an IBM Cloud Dedicated environment and supports provisioning the following resources

ocofaigh commented on solutions/dedicated/README.md:
```diff
@@ -0,0 +1,13 @@
+- An IBM Cloudant Dedicated instance on IBM Cloud Dedicated
+- A IBM Cloudant database
+
+![dedicated-cloudant](../../images/dedicated_cloudant.svg)
```
Can you add the diagram into a folder called `reference-architectures` instead. This is where they will need to be for when we publish a public DA.

rajatagarwal-ibm commented:
/run pipeline

rajatagarwal-ibm commented:
/run pipeline

ocofaigh commented:
/run pipeline

terraform-ibm-modules-ops commented:
:tada: This PR is included in version 1.1.1 :tada:

The release is available on [GitHub release](https://github.com/terraform-ibm-modules/terraform-ibm-cloudant/releases/tag/v1.1.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
